? .deps
? autom4te.cache
? bk-deps
? gtktripleclick.patch
? version-script
? wxgtk2d-2.5-config
? contrib/Makefile
? contrib/samples/Makefile
? contrib/samples/animate/Makefile
? contrib/samples/applet/Makefile
? contrib/samples/deprecated/Makefile
? contrib/samples/deprecated/proplist/Makefile
? contrib/samples/deprecated/resource/Makefile
? contrib/samples/deprecated/treelay/Makefile
? contrib/samples/fl/Makefile
? contrib/samples/gizmos/Makefile
? contrib/samples/gizmos/dynsash/Makefile
? contrib/samples/gizmos/dynsash_switch/Makefile
? contrib/samples/gizmos/editlbox/Makefile
? contrib/samples/gizmos/led/Makefile
? contrib/samples/gizmos/multicell/Makefile
? contrib/samples/gizmos/splittree/Makefile
? contrib/samples/mmedia/Makefile
? contrib/samples/ogl/Makefile
? contrib/samples/ogl/ogledit/Makefile
? contrib/samples/ogl/studio/Makefile
? contrib/samples/plot/Makefile
? contrib/samples/stc/Makefile
? contrib/samples/svg/Makefile
? contrib/samples/xrc/Makefile
? contrib/src/Makefile
? contrib/src/animate/Makefile
? contrib/src/applet/Makefile
? contrib/src/deprecated/Makefile
? contrib/src/fl/Makefile
? contrib/src/gizmos/Makefile
? contrib/src/mmedia/Makefile
? contrib/src/net/Makefile
? contrib/src/ogl/Makefile
? contrib/src/plot/Makefile
? contrib/src/stc/Makefile
? contrib/src/svg/Makefile
? contrib/src/xrc/Makefile
? contrib/utils/Makefile
? contrib/utils/convertrc/Makefile
? contrib/utils/wxrc/Makefile
? contrib/utils/wxrcedit/Makefile
? demos/Makefile
? demos/bombs/Makefile
? demos/dbbrowse/Makefile
? demos/forty/Makefile
? demos/fractal/Makefile
? demos/life/Makefile
? demos/poem/Makefile
? lib/wx
? samples/Makefile
? samples/artprov/Makefile
? samples/calendar/Makefile
? samples/caret/Makefile
? samples/checklst/Makefile
? samples/config/Makefile
? samples/console/Makefile
? samples/controls/Makefile
? samples/dialogs/Makefile
? samples/dialup/Makefile
? samples/dnd/Makefile
? samples/docview/Makefile
? samples/docvwmdi/Makefile
? samples/dragimag/Makefile
? samples/drawing/Makefile
? samples/dynamic/Makefile
? samples/erase/Makefile
? samples/event/Makefile
? samples/exec/Makefile
? samples/font/Makefile
? samples/grid/Makefile
? samples/help/Makefile
? samples/htlbox/Makefile
? samples/html/Makefile
? samples/html/about/Makefile
? samples/html/help/Makefile
? samples/html/helpview/Makefile
? samples/html/printing/Makefile
? samples/html/test/Makefile
? samples/html/virtual/Makefile
? samples/html/widget/Makefile
? samples/html/zip/Makefile
? samples/image/Makefile
? samples/internat/Makefile
? samples/joytest/Makefile
? samples/layout/Makefile
? samples/listctrl/Makefile
? samples/mdi/Makefile
? samples/menu/Makefile
? samples/minifram/Makefile
? samples/minimal/Makefile
? samples/notebook/Makefile
? samples/png/Makefile
? samples/printing/Makefile
? samples/propsize/Makefile
? samples/render/Makefile
? samples/rotate/Makefile
? samples/sashtest/Makefile
? samples/scroll/Makefile
? samples/scrollsub/Makefile
? samples/shaped/Makefile
? samples/sockets/Makefile
? samples/sound/Makefile
? samples/splitter/Makefile
? samples/statbar/Makefile
? samples/taskbar/Makefile
? samples/text/Makefile
? samples/thread/Makefile
? samples/toolbar/Makefile
? samples/treectrl/Makefile
? samples/typetest/Makefile
? samples/validate/Makefile
? samples/widgets/Makefile
? samples/wizard/Makefile
? utils/Makefile
? utils/HelpGen/Makefile
? utils/HelpGen/src/Makefile
? utils/configtool/Makefile
? utils/configtool/src/Makefile
? utils/emulator/Makefile
? utils/emulator/src/Makefile
? utils/helpview/Makefile
? utils/helpview/src/Makefile
? utils/hhp2cached/Makefile
? utils/tex2rtf/Makefile
? utils/tex2rtf/src/Makefile
Index: configure
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/configure,v
retrieving revision 1.669
diff -b -u -2 -r1.669 configure
--- configure	2004/02/10 21:49:18	1.669
+++ configure	2004/02/12 09:04:24
@@ -10782,5 +10782,6 @@
   (exit $ac_status); }; }; then
   for ac_declaration in \
-   '' \
+   ''\
+   '#include <stdlib.h>' \
    'extern "C" void std::exit (int) throw (); using std::exit;' \
    'extern "C" void std::exit (int); using std::exit;' \
@@ -10796,6 +10797,6 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-$ac_declaration
 #include <stdlib.h>
+$ac_declaration
 int
 main ()
@@ -11422,5 +11423,6 @@
 fi
 for ac_declaration in \
-   '' \
+   ''\
+   '#include <stdlib.h>' \
    'extern "C" void std::exit (int) throw (); using std::exit;' \
    'extern "C" void std::exit (int); using std::exit;' \
@@ -11436,6 +11438,6 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-$ac_declaration
 #include <stdlib.h>
+$ac_declaration
 int
 main ()
Index: src/gtk/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/gtk/window.cpp,v
retrieving revision 1.447
diff -b -u -2 -r1.447 window.cpp
--- src/gtk/window.cpp	2004/02/10 19:30:20	1.447
+++ src/gtk/window.cpp	2004/02/12 09:04:26
@@ -1538,4 +1538,19 @@
     wxEventType event_type = wxEVT_NULL;
 
+#ifdef __WXGTK20__
+    if (gdk_event->type == GDK_2BUTTON_PRESS && gdk_event->button >= 1 && 
+	gdk_event->button <= 3)
+    {
+        // Reset GDK internal timestamp variables in order
+        // to disable GDK triple click events. GDK will then
+        // next time believe no button has been clicked just before,
+        // and send a normal button click event.
+
+        GdkDisplay* display = gtk_widget_get_display (widget);
+        display->button_click_time[1] = 0;
+        display->button_click_time[0] = 0;
+    }
+#endif
+
     if (gdk_event->button == 1)
     {
@@ -1545,5 +1560,9 @@
         switch (gdk_event->type)
         {
-            case GDK_3BUTTON_PRESS:     // we could also map this to DCLICK...
+            // GDK_3BUTTON_PRESS events should not be generated by GDK 2.x
+            // since it is surpressed by code below. But for GDK 1.x we just
+            // map it to DOWN since I have not written surpressing code for GDK1.
+            case GDK_3BUTTON_PRESS:
+
             case GDK_BUTTON_PRESS:
                 event_type = wxEVT_LEFT_DOWN;
@@ -1563,4 +1582,5 @@
         switch (gdk_event->type)
         {
+            case GDK_3BUTTON_PRESS:
             case GDK_BUTTON_PRESS:
                 event_type = wxEVT_MIDDLE_DOWN;
@@ -1579,4 +1599,5 @@
         switch (gdk_event->type)
         {
+            case GDK_3BUTTON_PRESS:
             case GDK_BUTTON_PRESS:
                 event_type = wxEVT_RIGHT_DOWN;

 	  	 
